analyzers/tests/SonarAnalyzer.UnitTest/TestCases/SymbolicExecution/Roslyn/EmptyNullableValueAccess.cs

```diff
@@ -949,46 +949,99 @@ void Basics(MaybeInt i)
+            _ = i.Value;                                                     // Compliant, not a nullable type
         }
 
-        int ExplicitCast1 => ((StructWithValuePropertyAndCastOperators)(null as long?)).Value;                              // Noncompliant, FP, just gives 42
-        StructWithValuePropertyAndCastOperators ExplicitCast2 => (null as StructWithValuePropertyAndCastOperators?).Value;  // Noncompliant, FP, just gives a struct
-        int ExplicitCast3 => (null as StructWithValuePropertyAndCastOperators?).Value.Value;                                // Noncompliant, FP, just gives 42
+        int ExplicitCast1 => ((StructWithPropertyCalledValueAndCastOperators)(null as long?)).Value;                                    // Noncompliant, FP, just gives 42
+        StructWithPropertyCalledValueAndCastOperators ExplicitCast2 => (null as StructWithPropertyCalledValueAndCastOperators?).Value;  // Noncompliant, FP, just gives a struct
+        int ExplicitCast3 => (null as StructWithPropertyCalledValueAndCastOperators?).Value.Value;                                      // Noncompliant, FP, just gives 42
     }
 
-    class ClassWithValueProperty
+    class ClassWithPropertyCalledValue
     {
         public int Value => 42;
         public int APropertyNotCalledValue => 42;
     }
 
-    struct StructWithValuePropertyAndCastOperators
+    struct StructWithPropertyCalledValueAndCastOperators
     {
         public int Value => 42;
         public int APropertyNotCalledValue => 42;
 
-        public static implicit operator StructWithValuePropertyAndCastOperators(int? value) => new StructWithValuePropertyAndCastOperators();
-        public static explicit operator StructWithValuePropertyAndCastOperators(long? value) => new StructWithValuePropertyAndCastOperators();
+        public static implicit operator StructWithPropertyCalledValueAndCastOperators(int? value) => new StructWithPropertyCalledValueAndCastOperators();
+        public static explicit operator StructWithPropertyCalledValueAndCastOperators(long? value) => new StructWithPropertyCalledValueAndCastOperators();
     }
 }
 
+namespace TypeWithStaticPropertyCalledValue
+{
+    class Test
+    {
+        void Basics()
+        {
+            _ = ClassWithStaticPropertyCalledValue.Value;                                // Compliant, not on nullable value type
+            _ = ClassWithStaticPropertyCalledValue.Value.Value;                          // Compliant
+            _ = ClassWithStaticPropertyCalledValue.Value.Value.InstanceProperty;         // Compliant
+            _ = ClassWithStaticPropertyCalledValue.Value.Value.InstanceProperty.Value;   // Compliant
+            _ = new ClassWithInstancePropertyCalledValue().Value;                        // Compliant
+        }
+    }
+
+    class ClassWithStaticPropertyCalledValue
+    {
+        public ClassWithInstancePropertyCalledValue InstanceProperty => new ClassWithInstancePropertyCalledValue();
```

Comment:
They are there so that I can test nested `Value` calls: `Value.Value`, `Value.InstanceProperty.Value`, etc.

I couldn't use `new()`, and make the scaffolding less noisy because I get ``Feature 'target-typed object creation' is not available in C# 8.0. Please use language version 9.0 or greater.`, and I thought it was not appropriate to move the test to C#9 file just to use `new()`, because the thing being tested is not the `new()` syntax, but custom `Value` properties.

Makes sense?

Reply:
Nevermind, I got it, it does not make sense :-)

Updated.